app/Model/Event.php

```diff
@@ -1386,19 +1386,28 @@ private function __updateAttributeForSync($attribute, $server)
      * @param int $eventId
      * @param array $server
      * @param null|HttpSocket $HttpSocket
+     * @param boolean $metadataOnly, if True, we only retrieve the metadata
+     * without attributes and attachments which is much faster
      * @return array
      * @throws Exception
      */
-    public function downloadEventFromServer($eventId, $server, $HttpSocket=null)
+    public function downloadEventFromServer($eventId, $server, $HttpSocket=null, $metadataOnly=false)
     {
         $url = $server['Server']['url'];
         $HttpSocket = $this->setupHttpSocket($server, $HttpSocket);
         $request = $this->setupSyncRequest($server);
-        $uri = $url . '/events/view/' . $eventId . '/deleted[]:0/deleted[]:1/excludeGalaxy:1';
-        if (!empty($server['Server']['internal'])) {
-            $uri = $uri . '/excludeLocalTags:1';
+        if ($metadataOnly) {
+            $uri = $url . '/events/index';
```

Comment:
This is actually going to return a lot of data and will be rather heavy for what you need.

`$uri = $url . '/events/index/minimal:1';` is what you want.

Reply:
Ignore this, there's no need, missed the next line in that you restrict to an event id. The added benefit of minimal:1 is not worth it.